GDBMS_ALGO/community/label_prop.gsql

```diff
@@ -17,89 +24,164 @@ CREATE TEMPLATE QUERY GDBMS_ALGO.community.label_prop (SET<STRING> v_type_set, S
+                WHERE  s.@best_move.community == t.@community_id
+                       AND t.@to_change_community == TRUE
+                       AND t.@best_move.community == s.@community_id
+                       AND (s.@best_move.score < t.@best_move.score
+                            OR (abs(s.@best_move.score - t.@best_move.score) < 0.00000000001
+                                AND s.@vid > t.@vid))
+                POST-ACCUM
+                       s.@to_change_community = FALSE;
+    ChangedNodes = ChangedNodes MINUS SwapNodes;
 
-# Assign unique labels to each vertex
-Start = SELECT s 
-        FROM Start:s 
-        ACCUM s.@sum_label = getvid(s);
+    // Update community IDs
+    ChangedNodes = SELECT s
+                   FROM   ChangedNodes:s
+                   POST-ACCUM
+                          s.@community_id = s.@best_move.community,
+                          s.@to_change_community = FALSE;
 
-# Propagate labels to neighbors until labels converge or the max iterations is reached
-WHILE @@or_changed == true LIMIT maximum_iteration DO
-    @@or_changed = false;
-    Start = SELECT s 
-            FROM Start:s -(e_type_set:e)- :t
-            ACCUM t.@map += (s.@sum_label -> 1)  # count the occurrences of neighbor's labels
-            POST-ACCUM
-                INT max_v = 0,
-                INT label = 0,
-                # Iterate over the map to get the neighbor label that occurs most often
-                FOREACH (k,v) IN t.@map DO
-                    CASE WHEN v > max_v THEN
-                        max_v = v,
-                        label = k
-                    END
-                END,
-                # When the neighbor search finds a label AND it is a new label
-                # AND the label's count has increased, update the label.
-                CASE WHEN label != 0 AND t.@sum_label != label AND max_v > t.@sum_num THEN
-                    @@or_changed += true,
-                    t.@sum_label = label,
-                    t.@sum_num = max_v
-                END,
-                t.@map.clear();
-END;
+    // Find candidates for the next iteration
+    Candidates = SELECT t
+                 FROM   ChangedNodes:s -(e_type_set:e)- :t
+                 WHERE  t.@community_id != s.@community_id;
+  END;
 
-Start = {v_type_set};
-Start =  SELECT s 
-         FROM Start:s
-         POST-ACCUM 
-             IF result_attribute != "" THEN 
-                 s.setAttr(result_attribute, s.@sum_label) 
-             END,
-               
-             IF file_path != "" THEN 
-                 f.println(s, s.@sum_label) 
-             END,
-               
-             IF print_results THEN 
-                 @@comm_sizes_map += (s.@sum_label -> 1) 
-             END
-         LIMIT print_limit;
+  // Output results
+  Nodes = SELECT s
+          FROM All_Nodes:s
+          POST-ACCUM
+            IF result_attribute != "" THEN
+                s.setAttr(result_attribute, getvid(s.@community_id))
```

Comment:
@victorleeTG 
For file writing, I can handle both scenarios where there is only one vertex type and where there are multiple vertex types. However, for attribute writing, we have the following situations:

- One vertex type, the result_attribute is of type STRING
- Multiple vertex types, the result_attribute is of type STRING
- One vertex type, the result_attribute is of type INT
- Multiple vertex types, the result_attribute is of type INT

Is there a way to address these four situations?